src/info/manifest.c

```diff
@@ -383,6 +386,750 @@ manifestNewInternal(void)
     FUNCTION_TEST_RETURN(this);
 }
 
+/**********************************************************************************************************************************/
+typedef struct ManifestBuildData
+{
+    Manifest *manifest;
+    const Storage *storagePg;
+    const String *tablespaceId;                                     // Tablespace id if PostgreSQL version has one
+    bool online;                                                    // Is this an online backup?
+    const String *manifestWalName;                                  // Wal manifest name for this version of PostgreSQL
+    RegExp *dbPathExp;                                              // Identify paths containing relations
+    RegExp *tempRelationExp;                                        // Identify temp relations
+    RegExp *standbyExp;                                             // Identify files that must be copied from the primary
+    StringList *excludeContent;                                     // Exclude contents of directories
+    StringList *excludeSingle;                                      // Exclude a single file/link/path
+
+    // These change with each level of recursion
+    const String *manifestParentName;                               // Manifest name of this file/link/path's parent
+    const String *pgPath;                                           // Current path in the PostgreSQL data directory
+    bool dbPath;                                                    // Does this path contain relations?
+} ManifestBuildData;
+
+// Callback to process files/links/paths and add them to the manifest
+static void
+manifestBuildCallback(void *data, const StorageInfo *info)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM_P(VOID, data);
+        FUNCTION_TEST_PARAM(STORAGE_INFO, *storageInfo);
+    FUNCTION_TEST_END();
+
+    ASSERT(data != NULL);
+    ASSERT(info != NULL);
+
+    // Skip all . paths because they have already been recorded on the previous level of recursion
+    if (strEq(info->name, DOT_STR))
```

Comment:
Really? Where did we "record" this? I'm not sure I see where...

Reply:
`pg_data` is added to the manifest at line 890 and each path below is added at 472.